lib/Sema/CSSimplify.cpp

```diff
@@ -3543,19 +3543,21 @@ bool ConstraintSystem::repairFailures(
 
     // Drop the `tuple element` locator element so that all tuple element
     // mismatches within the same tuple type can be coalesced later.
+    auto index = elt.getAs<LocatorPathElt::TupleElement>()->getIndex();
```

Comment:
Just to double-check - this is an intermediate step until we get `ApplyArgToParam` generalization, right?

Reply:
Much of this is intermediate until we get a generalized `ApplyArgToParam` locator - especially the `getStructuralTypeContext` function in CSFix.cpp. That function is filling in the contextual information that we're missing.

That said, this particular line of code is just to keep track of which tuple elements/parameters don't match (it was there before, previously on line 3556, I just moved it up).

Reply:
Sounds good! I just needed to put this comment on one of the lines because otherwise GitHub doesn't support discussion if I just leave it as review comment :)